modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -331,6 +456,166 @@ public void onFailure(Exception e) {
         });
     }
 
+    /*
+     * This method executes the given force merge request. Once the request has completed successfully it writes a timestamp as custom
+     * metadata in the cluster state indicating when the force merge has completed. The listener is notified after the cluster state
+     * update has been made, or when the forcemerge fails or the write of the to the cluster state fails.
+     */
+    private void forceMergeIndex(ForceMergeRequest forceMergeRequest, ActionListener<Void> listener) {
+        assert forceMergeRequest.indices() != null && forceMergeRequest.indices().length == 1 : "DLM force merges one index at a time";
+        final String targetIndex = forceMergeRequest.indices()[0];
+        logger.info("DLM is issuing a request to force merge index [{}]", targetIndex);
+        client.admin().indices().forceMerge(forceMergeRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(ForceMergeResponse forceMergeResponse) {
+                logger.info("DLM successfully force merged index [{}]", targetIndex);
+                setForceMergeCompletedTimestamp(targetIndex, listener);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                if (e instanceof SnapshotInProgressException) {
```

Comment:
I don't think snapshotting an index causes forcemerge to fail (nor vice-versa). Is there a particular failure you have in mind here?

Reply:
I was blindly copying the delete code. Now that you mention it, it makes no sense. Oops. I'll remove that.